### ISSUE TITLE
strikethrough not working as expected ?

### ISSUE THREAD

**Issue by senges** [sphinx-contrib/confluencebuilder#771]
Hello ! Maybe I'm missing something, but I could not figure how to make inline strikethrough working, and no example usage seems to be provided in the documentation.

```
:strike:`example`  => ERROR: Unknown interpreted text role "strike".
~~example~~        => Interpreted as raw text
```

Only way I could make it work was :
```
.. role:: strike
    :class: strike

This is a :strike:`strikeme` example.
```

which is not very convenient. 

What am I missing here ? 
Sphinx / restructured text is still a bit obscure for me sometimes..

Thanks !

**Comment by senges** [comment]
relates to https://github.com/sphinx-contrib/confluencebuilder/pull/574

**Comment by jdknight** [comment]
reStructuredText does not define a standard `strike` role, and neither docutils or Sphinx provide an explicit role to support strikethrough text. This extension does not provide a `strike` role by default, to prevent any conflicts with other possible extensions or configurations may have that already define such a role.

The popular way to utilize strikethrough text is as you have referenced (repeating the above example):

```
.. role:: strike
    :class: strike

This is a :strike:`strikeme` example.
```

This is a common approach for HTML-based builds who wish to assign a `strike` class to text, and define their own appropriate style sheet (either themselves or provided from a theme) to help strikethrough the content. The referenced pull request in the previous comment was solely to help support environments which utilize such an approach (where this extension associates a `strike` class on text as an indication to strikethrough text).

For sure this may not be convenient defining the role in each document utilizing strikethroughs. Sphinx does provide a way to avoid this through its [`rst_prolog`](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-rst_prolog) configuration. For example, adding the following line in a project's `conf.py` file should provide the `strike` role to all documents for a project:

```
rst_prolog = """

.. role:: strike
    :class: strike

"""
```

On a side note, we could introduce a `confluence_strike` role which provides explicit support for Confluence-based strikethough text. This should prevent any chance of role collisions and provide a "ready" role for users.

**Comment by senges** [comment]
Thank you very much for you comprehensive explanation. I get the point, unfortunately the `rst_prolog` hack won't work for me as I'm using confluencebuilder with MyST plugin to parse markdown documentation (which does not support prologs / epilogs).

However, MyST has an extension called "strikethrough" that add `~~strike~~` syntax support for HTML output. Maybe I should dig in this direction. Or just accept the fact that I should redefine strike role on every single page of my documentation using strikethrough syntax for now.

Again thanks for the details, I'll come back with a workaround if I am able to find one in the next few days. If you don't have any additional suggestion regarding how to handle this situation, feel free to close !

**Comment by jdknight** [comment]
Took a look at MyST's `strikethrough` extension. It looks like [the implementation](https://github.com/executablebooks/MyST-Parser/blob/a784880c7fbd893566087b83deec081376740e2c/myst_parser/mdit_to_docutils/base.py#L1403-L1414) will create a raw `html` node for a Sphinx translator to process. However, the sphinxcontrib-confluencebuilder extension generates `confluence_storage` output, which in turn only officially supports raw nodes of a `confluence_storage` type.

If MyST's `strikethrough` extension is ever adjusted to create a custom strikethrough node type that can be translated on, the sphinxcontrib-confluencebuilder extension could create support for translating that node type. For now, there are two other approaches you could use:

### Using the Sphinx Role with MyST

There is a way to utilized Sphinx roles with MyST, which allows [`rst_prolog`](https://www.sphinx-doc.org/en/master/usage/configuration.html#confval-rst_prolog) to be used (if desired). Assuming the following configuration is set:

```
rst_prolog = """

.. role:: strike
    :class: strike

"""
```

reStructuredText would use the format:

```
This is a :strike:`strikeme` example.
```

And Markdown, the following can be used:

```
This is a {strike}`strikeme` example.
```

In this approach is used, the content will have a strikethrough on Confluence pages. Although for HTML-based content, a reminder that all this does it helps apply the `strike` CSS style for HTML-based content. If there is a need to support both builders, one would still require adding a custom CSS file to the configuration:

<sub>_[conf.py]_</sub>
```
html_static_path = ['_static']

def setup(app):
    app.add_css_file('css/test.css')
```

<sub>_[_static/css/test.css]_</sub>
```
.strike {
    text-decoration: line-through;
}
```

This should ensure both HTML and Confluence content is formatted as expected. The only side effect is that any common Markdown viewer/editors used for development may not gracefully show a render of the strikethrough.

### Permit Raw HTML Tags (unsupported)

An undocumented configuration is `confluence_adv_permit_raw_html`. When configured to `True`, any raw HTML nodes generated from Sphinx build will be added into the final Confluence content. Note that this configuration is completely unsupported, but may be a hack that you could be interested in. It seems to work with strikethroughs produced by MyST; it may cause issues for other content.

```
confluence_adv_permit_raw_html = True
```

**Comment by senges** [comment]
We had the exact same searching path !

I also had a look at the strikethrough extension, and indeed the MyST extension does generate HTML : `Strikethrough is currently only supported in HTML output`. That's why I dropped this solution for now.

Then I did the combo `rst_prolog` + custom CSS to make sure HTML output would also support the strike role. However, sphinx was still shouting at me `ERROR: Unknown interpreted text role "strike"`. Few issues (https://github.com/executablebooks/MyST-Parser/issues/270 for ex) suggest that MyST does not support `rst_prolog`, that's why I said this hack would not work here.

> Permit Raw HTML Tags (unsupported)

Alright I'll have a look at this, I was not aware this option even existed ! But as you said it looks a bit dangerous as I have many md files and it would take a while to make sure everything is fine using this flag.

**Comment by jdknight** [comment]
An overview of this issue should be as follows: reStructuredText and Sphinx do not have an official strikethrough capability. This extension does provide a somewhat common way to manage strikethroughs by applying the format when the `strike` class is detected for an inline node. However, the main use case for this issue appears to be focused on utilizing MyST's strikethrough extension, which at this time only supports HTML builds. If the MyST extension in the future reworks its strikethrough extension to generate a "strikethrough node", the Confluence builder extension could be modified to support the MyST strikethrough extension.

Since there is no any immediate action for Confluence builder extension to do, will mark this issue as closed. If MyST updates itself as mentioned above, or if we have possible better ways to manage strikethroughs with Confluence in the future, we can re-open this issue or create a new one.

In addition, a change has been introduced in this extension (https://github.com/sphinx-contrib/confluencebuilder/pull/784; available v2.1+) which will provide a Confluence-specific role to utilize strikethroughs (`confluence_strike`; limited to Confluence documents).
